internal/proposals/improvement-of-log-messages-usability.md

```diff
@@ -0,0 +1,75 @@
+# Improvement of log messages usability
+
+## Motivation
+
+We want to provide a common, unified solution to logging across Kyma components, which will enable users and maintainers to easily track specific events happening in Kyma's ecosystem. The solution should be independent of the platform (AWS, Azure, GCP) and easy to introduce in existing components
+
+Users, developers, and maintainers want to have an easy way to track events happening in Kyma, problems that occur, and debug faulty components. This will make understanding of Kyma's ecosystem easier and more transparent
+
+## Solution regarding the log messages
+
+Every mission-critical component (or a critical part of the component) must produce log entries in order to make the problem diagnosis process quick and effective
+
+Key assumptions for the log messages:
+
+- Transparency - every entry must be as short as possible but containing all essential information about the process
+- Tracing - entries sequence must be connected so there is an easy way to filter logs concerning a specific operation
+- GDPR-compliance - entries must not contain any sensitive data (password, personal data in the meaning of GDPR)
+- Agile - log messages and their context should be refactored. If the log was useful while debugging the incident, make other also that useful. If there was something missing in the log, report it and refactor
+- No "responsibility overlapping" - Because there are other tools to help in diagnosis, logs, in general, should not contain the information which can be taken from that tools (i.e there's tracing which can be used for timings)
+
+Additional assumptions:
+
+- If there is a quick solution for an error or simple suggestion for a warning, add it to the message so the operator could use it while debugging
```

Comment:
We need to be careful with such statements, this may lead to the log message flood if we log any single thing. I would indicate that such additions are applicable at the debug log level.

Reply:
Tru, done!